src/test/architecture/test_architecture.py

```diff
@@ -312,3 +313,579 @@ def test_verify_callback_uniqueness(self, mocker, caplog):
+        renamed_architecture_text = """
+named_paths:
+- path_name: target_path_0
+  node_chain:
+  - node_name: /node_0
+    publish_topic_name: /topic_0
+  - node_name: /changed_node
+    subscribe_topic_name: /topic_0
+- path_name: target_path_1
+  node_chain:
+  - node_name: /changed_node
+    publish_topic_name: /topic_1
+  - node_name: /node_0
+    subscribe_topic_name: /topic_1
+executors:
+- executor_type: single_threaded_executor
+  executor_name: executor_0
+  callback_group_names:
+  - /callback_group_0
+- executor_type: single_threaded_executor
+  executor_name: executor_1
+  callback_group_names:
+  - /callback_group_1
+nodes:
+- node_name: /node_0
+  callback_groups:
+  - callback_group_type: mutually_exclusive
+    callback_group_name: /callback_group_1
+    callback_names:
+    - /callback_0
+    - /callback_1
+  callbacks:
+  - callback_name: /callback_0
+    callback_type: timer_callback
+    period_ns: 200000000
+    symbol: timer
+  - callback_name: /callback_1
+    callback_type: subscription_callback
+    topic_name: /topic_1
+    symbol: sub
+  publishes:
+  - topic_name: /topic_0
+    callback_names:
+    - /callback_0
+  subscribes:
+  - topic_name: /topic_1
+    callback_name: /callback_1
+- node_name: /changed_node
+  callback_groups:
+  - callback_group_type: mutually_exclusive
+    callback_group_name: /callback_group_0
+    callback_names:
+    - /callback_2
+    - /callback_3
+  callbacks:
+  - callback_name: /callback_2
+    callback_type: timer_callback
+    period_ns: 200000000
+    symbol: timer
+  - callback_name: /callback_3
+    callback_type: subscription_callback
+    topic_name: /topic_0
+    symbol: sub
+  publishes:
+  - topic_name: /topic_1
+    callback_names:
+    - /callback_2
+  subscribes:
+  - topic_name: /topic_0
+    callback_name: /callback_3"""
+
+        mocker.patch('builtins.open', mocker.mock_open(read_data=self.renaming_architecture_text))
+        arch = Architecture('yaml', 'architecture.yaml')
+        assert arch.get_node('/node_0')
+        assert arch.get_node('/node_1')
+        with pytest.raises(ItemNotFoundError) as e:
+            assert arch.get_node('/changed_node')
+        assert e
+
+        arch.rename_node('/node_1', '/changed_node')
+        assert arch.get_node('/node_0')
+        assert arch.get_node('/changed_node')
+        with pytest.raises(ItemNotFoundError) as e:
+            assert arch.get_node('/node_1')
+        assert e
+
+        mocker.patch('builtins.open', mocker.mock_open(read_data=renamed_architecture_text))
+        arch_expected = Architecture('yaml', 'architecture.yaml')
+
+        assert set(arch_expected.node_names) == set(arch.node_names)
+
+        # check this
+        # - path_name: target_path_0
+        #   node_chain:
+        #   - node_name: /node_0
+        #   - node_name: /changed_node
+        assert set(arch_expected.get_path('target_path_0').node_names) \
+            == set(arch.get_path('target_path_0').node_names)
```

Comment:
Because the return value of anitecture object is a ValueObject, it can be evaluated as follows.
```suggestion
        assert arch.nodes == arch_expected.nodes
        assert arch.communications == arch_expected.communications
        assert arch.paths == arch_expected.paths
        assert arch.executors == arch_expected.executors
```

Reply:
Fixed it in [370ca2d](https://github.com/tier4/CARET_analyze/pull/156/commits/370ca2da6a861c0b72cd6f1cfdded510c16ec622)